app/Http/Controllers/MovieController.php

```diff
@@ -2,9 +2,74 @@
 
 namespace App\Http\Controllers;
 
+use Carbon\Carbon;
+use App\Models\Movie;
 use Illuminate\Http\Request;
 
 class MovieController extends Controller
 {
-    //
+    public function index() //print all data in Moives database
+    {
+        return view('movies.index', [
+            'movies' => Movie::all()
+        ]);
+    }
+
+    public function show(Movie $movie)
+    {
+        return view('movies.show', compact('movie'));
+        // similar to return view('movies.show', $movie=>movie);
+    }
+
+    public function create()
+    {
+        return view('movies.create');
+    }
+
+    public function store(Request $request)
+    {
+        $request->validate([
+            'title' => ['required', 'string', 'min:3'],
+            'release_date' => ['required', 'date'],
+            'poster' => ['required', 'string', 'min:16'],
+            'trailer' => ['required', 'string', 'min:16'],
+            'blurb' => ['required', 'string']
+        ]);
+
+        Movie::create([
+            'title' => request()->input('title'),
+            'release_date' => Carbon::parse(request()->input('release_date'))->format('d-m-Y'),
+            'poster' => request()->input('poster'),
+            'trailer' => request()->input('trailer'),
+            'blurb' => request()->input('trailer')
+        ]);
+
+        return redirect()->route('movies.index')->with('success', 'Movie added');
+
+    }
+
+    public function edit(Movie $movie)
+    {
+        return view('movies.edit', compact('movie'));
+    }
+
+    public function update(Request $request, Movie $movie)
+    {
+        $movie->update($request->validate([
+            'title' => ['required', 'string', 'min:3'],
+            'release_date' => ['required', 'date'],
+            'poster' => ['required', 'string', 'min:16'],
+            'trailer' => ['required', 'string', 'min:16'],
+            'blurb' => ['required', 'string']
+        ]));
```

Comment:
You can add a blank line before the return 😄 , this makes the code a bit cleaner

Reply:
I believe for this one you still want to have the release date saving in the correct format. So for now, just do the same you have done in the `store()` method. 😸

Reply:
Changes made